bulk-ingester/src/test/java/org/alfresco/hxi_connector/bulk_ingester/processor/BulkIngestionProcessorIntegrationTest.java

```diff
@@ -0,0 +1,203 @@
+
+        // when
+        bulkIngestionProcessor.process();
+
+        // then
+        testNodePublisher.assertPublishedNodes(categories);
+        testNodePublisher.assertPublishedNode(folder);
+        testNodePublisher.assertPublishedNode(textFile);
+        testNodePublisher.assertPublishedNodes(pdfFiles);
+    }
+
+    @TestConfiguration
+    public static class MockEventPublisherConfiguration
+    {
+        @Bean
+        @Primary
+        public TestNodePublisher nodePublisher()
+        {
+            return new TestNodePublisher();
+        }
+
+        @Bean
+        @Primary
+        public ApplicationManager applicationManager()
+        {
+            return mock();
+        }
+    }
+
+    @Slf4j
+    public static class TestNodePublisher implements NodePublisher
+    {
+        private final Map<String, Node> publishedNodes = new HashMap<>();
+
+        @Override
+        public void publish(Node node)
+        {
+            log.info("Publishing node {}", node.toString());
+
+            publishedNodes.put(node.nodeId(), node);
+        }
+
+        public void assertPublishedNodes(List<Node> nodes)
+        {
+            nodes.forEach(this::assertPublishedNode);
+        }
+
+        public void assertPublishedNode(Node node)
+        {
+            Node publishedNode = publishedNodes.get(node.nodeId());
+
+            assertNotNull(publishedNode, format("Node %s not published", node.nodeId()));
+
+            assertEquals(node.nodeId(), publishedNode.nodeId());
+            assertEquals(node.contentInfo(), publishedNode.contentInfo());
+            assertMapsEquals(node.properties(), publishedNode.properties());
+        }
+    }
+
+    private static Map<String, Serializable> parseProperties(String... properties)
+    {
+        return Arrays.stream(properties)
+                .map(BulkIngestionProcessorIntegrationTest::parseProperty)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private static Map.Entry<String, Serializable> parseProperty(String property)
+    {
+        String[] propertySplit = property.split("=");
+
+        String key = propertySplit[0];
+
+        if (propertySplit.length == 1)
+        {
+            return Map.entry(key, "");
+        }
+
+        String value = propertySplit[1];
+
+        return Map.entry(key, parsePropertyValue(value));
+    }
+
+    private static Serializable parsePropertyValue(String value)
+    {
+        if (value.startsWith("["))
+        {
+            value = value.replace("[", "");
+            value = value.replace("]", "");
+            value = value.replace(" ", "");
```

Comment:
Are you doing this because the list has spaces next to the commas? Can we split with `value.split(", ")` instead?